frontend/lib/src/components/widgets/Slider/Slider.tsx

```diff
@@ -282,159 +149,294 @@ class Slider extends React.PureComponent<Props, State> {
         }
       }
     }
-  }
+  })
+
+  // Update the thumb numbers via DOM manipulation to avoid a redraw, which
+  // drops the widget's focus state.
+  useEffect(() => {
+    thumbValueRefs.map((ref, i) => {
+      if (!ref.current) {
+        return
+      }
+      ref.current.innerText = formattedValueArr[i]
+    })
 
-  // eslint-disable-next-line react/display-name
-  private renderThumb = React.forwardRef<HTMLDivElement, StyleProps>(
-    (props: StyleProps, ref): JSX.Element => {
-      const { $value, $thumbIndex } = props
-      const thumbIndex = $thumbIndex || 0
-      this.thumbRef[thumbIndex] = ref as React.MutableRefObject<HTMLDivElement>
-      this.thumbValueRef[thumbIndex] ||= React.createRef<HTMLDivElement>()
-
-      const formattedValue = $value
-        ? this.formatValue($value[$thumbIndex as number])
-        : ""
-      const passThrough = pick(props, [
-        "role",
-        "style",
-        "aria-valuemax",
-        "aria-valuemin",
-        "aria-valuenow",
-        "tabIndex",
-        "onKeyUp",
-        "onKeyDown",
-        "onMouseEnter",
-        "onMouseLeave",
-        "draggable",
-      ])
-      const ariaValueText: Record<string, string> = {}
-
-      if (this.props.element.options.length > 0 || this.isDateTimeType()) {
-        ariaValueText["aria-valuetext"] = formattedValue
+    thumbRefs.map((ref, i) => {
+      if (!ref.current) {
+        return
       }
+      ref.current.setAttribute("aria-valuetext", formattedValueArr[i])
+    })
+  })
 
-      return (
-        <StyledThumb
-          {...passThrough}
-          disabled={props.$disabled === true}
-          ref={this.thumbRef[thumbIndex]}
-          aria-valuetext={formattedValue}
-          aria-label={this.props.element.label}
-        >
-          <StyledThumbValue
-            data-testid="stSliderThumbValue"
-            disabled={props.$disabled === true}
-            ref={this.thumbValueRef[thumbIndex]}
-          >
-            {formattedValue}
-          </StyledThumbValue>
-        </StyledThumb>
-      )
+  // When resetting a form, `value` will change so we need to change `uiValue`
+  // to match.
+  useEffect(() => {
+    if (value !== uiValue) {
+      setUiValue(value)
     }
+    // Don't include `uiValue` in the deps below or the slider will become
+    // jittery.
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [value])
```

Comment:
According to the [React docs](https://react.dev/reference/react/useState#setstate-caveats) setting the state to the same value does not trigger a re-render:
> If the new value you provide is identical to the current state, as determined by an [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will skip re-rendering the component and its children. This is an optimization. Although in some cases React may still need to call your component before skipping the children, it shouldn’t affect your code.

Based on this, I would remove the `if (value !== uiValue)` check and, thus, remove disabling eslint 🙂 

```typescript
  useEffect(() => {
    setUiValue(value)
  }, [value])
```